M2/Macaulay2/packages/Varieties.m2

```diff
@@ -223,6 +287,25 @@ OO = new ScriptedFunctor from {
 OO.texMath = ///{\mathcal O}///
 installMethod(symbol_, OO, Variety, (OO, X) -> sheaf(X, ring X))
 
+isWellDefined CoherentSheaf := F -> (
+    M := module F;
+    X := variety F;
+    true -- TODO: isWellDefined M and isWellDefined X
+    -- data type checks
+    and condition(set keys F === set { symbol variety, symbol module, symbol cache },
+	"the hash table does not have the expected keys")
+    and condition(
+	instance(F.variety, Variety) and
+	instance(F.module, Module)   and
+	instance(F.cache, CacheTable),
+	"the hash table does not have the expected values")
+    -- mathematical checks
+    and condition(ring M === ring X,
+	"underlying module and variety do not have the same ring")
+    and condition(not isProjective X or isHomogeneous M,
+	"underlying module of coherent sheaf on a projective variety should be homogeneous")
```

Comment:
Is the ```isHomogeneous M``` sufficient?  What sort of gradings do/should we allow the ```Proj``` method to handle?

Reply:
I believe so? Do you have an example in mind?
It's certainly possible that we need more assertions for well-defined-ness of coherent sheaves though.